src/main.tf

```diff
@@ -19,4 +19,5 @@ resource "azurerm_subnet" "main" {
   resource_group_name  = azurerm_resource_group.main.name
   virtual_network_name = azurerm_virtual_network.main.name
   address_prefixes     = [cidrsubnet(var.cidr, 1, 0)]
+  service_endpoints    = ["Microsoft.Storage", "Microsoft.Sql", "Microsoft.KeyVault", "Microsoft.ContainerRegistry"]
```

Comment:
No worries on the dynamic nature of it. More importantly, we try not to enable things that we don't need. I understand why some of these might make sense to include like SQL and whatnot but why not include `AzureCosmosDB` as well? etc...

The full list of options
```
Microsoft.AzureActiveDirectory, Microsoft.AzureCosmosDB, Microsoft.ContainerRegistry, 
Microsoft.EventHub, Microsoft.KeyVault, Microsoft.ServiceBus, Microsoft.Sql, 
Microsoft.Storage and Microsoft.Web
```

This should be `service_endpoints = ["Microsoft.Storage"]` unless we do need the others. 👍

Reply:
`AzureCosmosDB` bundles create their own subnet, so they're not needed. I enabled `Sql` because that supports Microsoft's `Azure SQL DB` serverless service, which we haven't created yet, so I'm not sure whether or not we'll be creating it's own subnet. I can just do storage for now, however with every new type we have a bundle for going forward, we'll need to add a service endpoint to the bundle and users will need to redeploy their VNet manifests.

Reply:
The forward-thinking is good and I get what you mean by having to add and redeploy but let's keep this change small and scoped and come back when those use cases crystalize.